backend/application/syson-application-configuration/src/main/java/org/eclipse/syson/application/publication/SySONLibraryPublicationHandler.java

```diff
@@ -0,0 +1,258 @@
+                        List.of(new Message("Library '%s' (version '%s') already exists in namespace '%s'.".formatted(libraryName, libraryVersion, libraryNamespace), MessageLevel.ERROR)));
+            }
+        } else {
+            // There are no proper SysML contents to publish.
+            result = new ErrorPayload(input.id(),
+                    List.of(new Message("There are no SysML contents in project '%s' to publish as library.".formatted(project.getName()), MessageLevel.ERROR)));
+        }
+
+        return result;
+    }
+
+    private Resource createResource(final String resourceName, final String resourceId, final Collection<EObject> contents) {
+        final JSONResourceFactory jsonResourceFactory = new JSONResourceFactory();
+        final URI uri = jsonResourceFactory.createResourceURI(UUID.nameUUIDFromBytes(resourceId.getBytes()).toString());
+        final Resource isolatedResource = jsonResourceFactory.createResource(uri);
+        isolatedResource.eAdapters().add(new ResourceMetadataAdapter(resourceName));
+        isolatedResource.getContents().addAll(contents);
+        return isolatedResource;
+    }
+
+    protected Optional<SemanticData> publishAsLibrary(final ICause parentCause, final ResourceSet resourceSet, final String name, final String version,
+            final List<AggregateReference<SemanticData, UUID>> dependencies) {
+        final ICause cause = new SysONPublishedLibrarySemanticDataCreationRequested(parentCause, name);
+        return this.createSemanticData(cause, resourceSet.getResources(), dependencies);
+    }
+
+    protected Optional<SemanticData> createSemanticData(final ICause event, final Collection<Resource> resources, final List<AggregateReference<SemanticData, UUID>> dependencies) {
+        final List<DocumentData> documentDatas = resources.stream().map(resource -> this.resourceToDocumentService.toDocument(resource, false)).filter(Optional::isPresent).map(Optional::get).toList();
+        final List<Document> documents = documentDatas.stream().map(DocumentData::document).toList();
+        final List<String> domains = documentDatas.stream().map(DocumentData::ePackageEntries).flatMap(List::stream).map(EPackageEntry::nsURI).distinct().toList();
+
+        final IResult<SemanticData> creationResult = this.semanticDataCreationService.create(event, documents, domains, dependencies);
+        if (creationResult instanceof Success<SemanticData> creationSuccess) {
+            return Optional.ofNullable(creationSuccess.data());
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    protected Map<Resource, List<EObject>> getProperSysMLRootContents(final ResourceSet resourceSet) {
+        Objects.requireNonNull(resourceSet);
+
+        final List<Resource> dependencies = this.getDependenciesToPublishedLibraries(resourceSet);
+        final List<Resource> candidateResources = resourceSet.getResources().stream().filter(resource -> !dependencies.contains(resource)).toList();
```

Comment:
Why creating a list here? You could avoid that and do the `.forEach(resource ...` just after the `.filter(resource -> !dependencies.contains(resource))`